src/shared/redux/modules/agreedSample.ts

```diff
@@ -12,35 +11,70 @@ const AGREED_SAMPLE_GET_TEXT_SUCCESS =
   "redux-proto/agreedsample/get-text-success";
 const AGREED_SAMPLE_GET_TEXT_FAIL = "redux-proto/agreedsample/get-text-fail";
 
-interface GetTextRequestAction {
+type GetTextRequestPayload = {
+  resource: string;
+  params: { status?: string | null };
+};
```

Comment:
微妙なんだよなぁ、これ。このactionでこのtypeが必要なわけじゃなくて、fetchrReadでpayloadがこの形になってる必要があるだけなんだよね。だからここでは正直payloadを設定する必要あるのかなーとは思ってる。

最初にkoichikが書いたのがこの形だったから、割と踏襲されてるんだけど、action payloadがこうじゃないといけないわけじゃない。

最終的に必要なのは、 fetchrReadの呼び出し方でそこに型があれば、ここはむしろactionとかなくても良いんじゃないかなと。

Reply:
ジェネリックなActionが全てその形なら問題だけど、Fetchrミドルウェアに喰わせるのが前提のActionならそれで構わないと思います。
このモジュール (ducksなヤツ) の利用者、典型的にはContainerコンポーネントですが、それに対する公開インタフェースはAction Creatorという関数であり、ActionオブジェクトとりわけPayloadの構造は実装の詳細という認識です。
なのでFetchrに依存したPayloadの型があること自体はそれでよいかと。

しかし、実装詳細であるPayloadの構造がモジュールの利用者にダダ漏れになるのは好ましくないと思います。
利用者に対してPayloadは不透明 (Opaque) であるべき。
一方でモジュールの内部、とりわけReducerなどはPayloadの構造に対してタイプセーフであるべきなので、型変数やデフォルト値を活用するなどして両立できるといいですね。

Reply:
元々は以下のように直前のstepの `ResolvedType` を、直後のstepの引数にとりたかったので今回の修正を入れていました。
```ts
  function steps<S, F, A = any, B = any>(
    s0: PromisableType<A>,
    s1: (a: ResolvedType<A>) => PromisableType<B>,
    end: [
      (b: ResolvedType<B>) => PromisableType<S>,
      (err: any) => PromisableType<F>
    ],
  ): Promise<S> | Promise<F>;
```

ただ redux-effects-stepsの定義は下記のため、 `「各stepで任意の引数をとれるように型定義を更新しても良いかもね」` という話し合いはしました。

## API

## Action Creators

#### `steps(origin, [...steps])`

##### Arguments

* `origin` *(Array|Object)*
    * One of the:
        * `action` *(Object)*: A 
          [FSA](https://github.com/acdlite/flux-standard-action)
          compliant action object.
        * `actions` *(Array)*: An array of actions.
* `steps` *(Array)*
    * Each element is in a `[success, failure]` tuple Array.
    * `success` and `failure` are one of the:
        * `action` *(Object)*: A
          [FSA](https://github.com/acdlite/flux-standard-action)
          compliant action object.
        * `actions` *(Array)*: An array of
          [FSA](https://github.com/acdlite/flux-standard-action)
          compliant action object.
        * `actionCreator` *(Function)*: An action creator.
    * If `failure` is not needed, you can pass just `success` instead of an Array.

Reply:
上の自分のコメントは勘違いでした。
xxxRequestなアクションがFetchrへのActionと同じペイロードを持っているのはRedux DevToolsで調べやすくするためです。
`fetchRead(...)` なアクションは redux-effects-fetchr で消費されてしまうため、Redux DevToolsに出てきません。そのため、どんなAPIリクエストが呼ばれるのか確認しにくいという指摘が (確か峯本さんから) あり、その対策として xxxRequest アクションにも同じペイロードを持たせるようにしたという経緯がありました。
なので絶対に必要ということはありませんが、実用上の利便性が高まると思います。